src/bindings/gaiabuild/legacy/env.js

```diff
@@ -2,62 +2,45 @@
 
 import { L10nError } from '../../../lib/errors';
 import { Env, amendError } from '../../../lib/env';
+import { LegacyContext } from './context';
 import { createEntry } from './resolver';
 import PropertiesParser from './parser';
 import { walkContent } from './pseudo';
 import { qps } from '../../../lib/pseudo';
-import {
-  emit, addEventListener, removeEventListener
-} from '../../../lib/events';
 
-// XXX babel transpiles class inheritance to code which modifies prototypes 
-// which triggers warnings in Gecko;  we redefine LegacyEnv from scratch to 
-// avoid warnings.
-export class LegacyEnv {
-  constructor(defaultLang, fetch) {
-    // XXX babel doesn't allow calling the constructor as a regular function 
-    // so we need to copy the whole constructor from Env
```

Comment:
I found a workaround for this:  using `LegacyEnv.prototype = Object.create(Env.prototype);`, `this` has `Env` in its prototypes chain, so babel's `classCallCheck` doesn't complain anymore.